src/parser.ts

```diff
@@ -619,19 +622,21 @@ export class Parser extends DiagnosticEmitter {
       return null;
     }
     // ... | null
-    while (tn.skip(Token.BAR)) {
-      if (tn.skip(Token.NULL)) {
-        type.isNullable = true;
-      } else {
-        let notNullStart = tn.pos;
-        let notNull = this.parseType(tn, false, true);
-        if (!suppressErrors) {
-          this.error(
-            DiagnosticCode._0_expected,
-            notNull ? notNull.range : tn.range(notNullStart), "null"
-          );
+    if (tn.peek() != Token.BAR_BAR) {
+      while (tn.skip(Token.BAR)) {
```

Comment:
With `BAR_BAR` and `BAR` being distinct tokens (`BAR_BAR` is not two times `BAR`), this change seems like it is not doing much. Or am I missing something? :)

Reply:
After following without this check I got errors for case some `a || b` expressions in non-type context
<img width="499" alt="screenshot" src="https://user-images.githubusercontent.com/1301959/145181432-bc114add-5aa8-4041-a8e6-1b2af929d381.png">
.

Reply:
That's strange. How would code in `parseType`, where `||` is invalid, affect expressions?

Reply:
Yes, I was surprised too. You could try comment that `BAR_BAR` check and try to run compile tests to see that strange behaviour